x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/EnrichPolicyResolver.java

```diff
@@ -61,105 +86,344 @@ public EnrichPolicyResolver(ClusterService clusterService, TransportService tran
+            case REMOTE -> includeLocal
+                ? CollectionUtils.appendToCopy(remoteClusters, RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY)
+                : remoteClusters;
+        };
+    }
+
+    /**
+     * Resolve an enrich policy by merging the lookup responses from the target clusters.
+     * In ANY mode, the exact enrich fields across policies are required,
+     * while in REMOTE mode, a superset of enrich fields is used for the combined policy.
+     * @return a resolved enrich policy or an error
+     */
+    private Tuple<ResolvedEnrichPolicy, String> mergeLookupResults(
+        UnresolvedPolicy unresolved,
+        Collection<String> targetClusters,
+        Map<String, LookupResponse> lookupResults
+    ) {
+        assert targetClusters.isEmpty() == false;
+        String policyName = unresolved.name;
+        final Map<String, ResolvedEnrichPolicy> policies = new HashMap<>();
+        final List<String> failures = new ArrayList<>();
+        for (String cluster : targetClusters) {
+            LookupResponse lookupResult = lookupResults.get(cluster);
+            if (lookupResult != null) {
+                ResolvedEnrichPolicy policy = lookupResult.policies.get(policyName);
+                if (policy != null) {
+                    policies.put(cluster, policy);
+                } else {
+                    final String failure = lookupResult.failures.get(policyName);
+                    if (failure != null) {
+                        failures.add(failure);
                     }
                 }
-            }), LookupResponse::new, threadPool.executor(EsqlPlugin.ESQL_THREAD_POOL_NAME))
+            }
+        }
+        if (targetClusters.size() != policies.size()) {
+            final String reason;
+            if (failures.isEmpty()) {
+                List<String> missingClusters = targetClusters.stream().filter(c -> policies.containsKey(c) == false).sorted().toList();
+                reason = missingPolicyError(policyName, targetClusters, missingClusters, lookupResults);
+            } else {
+                reason = "failed to resolve enrich policy [" + policyName + "]; reason " + failures;
+            }
+            return Tuple.tuple(null, reason);
+        }
+        Map<String, EsField> mappings = new HashMap<>();
+        Collection<String> enrichFields = new LinkedHashSet<>();
+        Map<String, String> concreteIndices = new HashMap<>();
+        ResolvedEnrichPolicy last = null;
+        for (Map.Entry<String, ResolvedEnrichPolicy> e : policies.entrySet()) {
+            ResolvedEnrichPolicy curr = e.getValue();
+            if (last != null && last.matchField().equals(curr.matchField()) == false) {
+                String error = "enrich policy [" + policyName + "] has different match_field across clusters";
+                return Tuple.tuple(null, error);
+            }
+            if (last != null && last.matchType().equals(curr.matchType()) == false) {
```

Comment:
We can support different match_type for the REMOTE mode, but I don't think we should.